src/cow-react/modules/limitOrders/containers/RateInput/index.tsx

```diff
@@ -54,16 +66,22 @@ export function RateInput() {
   }
 
   const isDisabledMPrice = useMemo(() => {
-    return isLoadingExecutionRate || !executionRate || activeRate?.equalTo(executionRate)
-  }, [activeRate, executionRate, isLoadingExecutionRate])
+    if (isLoadingExecutionRate) return true
+
+    if (executionRate) {
+      return activeRate?.equalTo(executionRate)
+    } else {
+      return !!initialRate && activeRate?.equalTo(initialRate)
+    }
+  }, [activeRate, executionRate, isLoadingExecutionRate, initialRate])
 
   return (
     <styledEl.Wrapper>
       <styledEl.Header>
         <HeadingText currency={primaryCurrency} rateImpact={rateImpact} />
 
         <styledEl.MarketPriceButton disabled={isDisabledMPrice} onClick={handleSetMarketPrice}>
-          <span>Market price</span>
+          <span>{executionRate ? 'Market price' : 'Update price'}</span>
```

Comment:
small question with label https://cowservices.slack.com/archives/C0361CDG8GP/p1669823385058719?thread_ts=1669818427.372219&cid=C0361CDG8GP

Reply:
Answered. Removed this